src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5108,7 +5155,7 @@ private static bool CanReuseVariableDeclarator(CSharp.Syntax.VariableDeclaratorS
             // declaration.  However, in the specific case of variable declarators, Dev10
             // specifically treats it as a variable name, even if it could be interpreted as a
             // keyword.
-            var name = this.ParseIdentifierToken();
+            var name = this.ParseIdentifierToken(reportErrorRatherThanConvertingFieldOrValue: true);
```

Comment:
for somethign as core as ParseIdentifierToken, it feels *very* strange to need such a specialized arg.  esp.  since we haven't needed that sort of arg for other contextual tokens?